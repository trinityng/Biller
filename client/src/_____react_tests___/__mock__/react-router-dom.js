import React from 'react';
// Just render plain div with its children
const rrd = require('react-router-dom');

rrd.BrowserRouter = ({children}) => <div>{children}</div>
module.exports = rrd;