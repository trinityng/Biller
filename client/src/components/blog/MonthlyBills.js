import React from 'react';

import { Row, Col, Card, CardHeader, CardBody, Button } from 'shards-react';

import RangeDatePicker from '../common/RangeDatePicker';
import Chart from '../../utils/chart';

class MonthlyBills extends React.Component {
  constructor(props) {
    super(props);
    this.canvasRef = React.createRef();
  }

  renderChart = () => {
    const MonthlyExpenses = new Chart(this.canvasRef.current, {
      type: 'LineWithLine',
      data: this.props.data,
      options: this.props.chartOptions,
    });
    if (this.props.data.datasets > 1) {
      // They can still be triggered on hover.
      const buoMeta = MonthlyExpenses.getDatasetMeta(0);

      buoMeta.data[0]._model.radius = 0;
      buoMeta.data[
        this.props.data.datasets[0].data.length - 1
      ]._model.radius = 0;

      // Render the chart.
      MonthlyExpenses.render();
    }
  };

  componentDidUpdate() {
    this.renderChart();
  }

  componentDidMount() {
    this.renderChart();
  }

  render() {
    return (
      <Card small className="h-100">
        <CardHeader className="border-bottom">
          <h6 className="m-0">Monthly Expenses</h6>
        </CardHeader>
        <CardBody className="pt-0">
          <Row className="border-bottom py-2 bg-light">
            <Col sm="6" className="d-flex mb-2 mb-sm-0">
              <RangeDatePicker />
            </Col>
            <Col>
              <Button
                size="sm"
                className="d-flex btn-white ml-auto mr-auto ml-sm-auto mr-sm-0 mt-3 mt-sm-0"
              >
                View Full Report &rarr;
              </Button>
            </Col>
          </Row>
          <canvas
            height="120"
            ref={this.canvasRef}
            style={{ maxWidth: '100% !important' }}
          />
        </CardBody>
      </Card>
    );
  }
}

export default MonthlyBills;
