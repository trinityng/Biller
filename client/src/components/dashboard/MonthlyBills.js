import React from 'react';
import { Card, CardHeader, CardBody } from 'shards-react';
import Chart from '../../utils/chart';

class MonthlyBills extends React.Component {
  constructor(props) {
    super(props);
    this.canvasRef = React.createRef();
  }

  renderChart = () => {
    const MonthlyExpenses = new Chart(this.canvasRef.current, {
      type: 'LineWithLine',
      data: this.props.data,
      options: this.props.chartOptions,
    });
    if (this.props.data.datasets > 1) {
      // They can still be triggered on hover.
      const buoMeta = MonthlyExpenses.getDatasetMeta(0);

      buoMeta.data[0]._model.radius = 0;
      buoMeta.data[
        this.props.data.datasets[0].data.length - 1
      ]._model.radius = 0;

      // Render the chart.
      MonthlyExpenses.render();
    }
  };

  componentDidUpdate() {
    this.renderChart();
  }

  componentDidMount() {
    this.renderChart();
  }

  render() {
    return (
      <Card small className="h-100">
        <CardHeader className="border-bottom">
          <h6 className="m-0">Monthly Expenses</h6>
        </CardHeader>
        <CardBody className="pt-0">
          <canvas
            height="120"
            ref={this.canvasRef}
            style={{ maxWidth: '100% !important' }}
          />
        </CardBody>
      </Card>
    );
  }
}

export default MonthlyBills;
