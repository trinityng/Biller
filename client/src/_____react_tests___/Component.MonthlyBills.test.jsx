import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import {Provider} from 'react-redux';
import {mount, shallow, configure } from 'enzyme';
import { MemoryRouter, Router } from "react-router-dom";
import React from 'react';
import Adapter from 'enzyme-adapter-react-16';

import MonthlyBills from '../components/dashboard/MonthlyBills';


Object.defineProperty(window, 'matchMedia', {
  writable: true,
  value: jest.fn().mockImplementation(query => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(), // deprecated
    removeListener: jest.fn(), // deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })),
});

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

configure({adapter: new Adapter()});

describe('MonthlyBills components', () => {
    let initialState = {};
    let store;

    const dummy_data = {
        labels: "chartLabels",
        datasets: [
          {
            label: 'Expenses',
            fill: 'start',
            data: [],
            backgroundColor: 'rgba(255,65,105,0.1)',
            borderColor: 'rgba(255,65,105,1)',
            pointBackgroundColor: '#ffffff',
            pointHoverBackgroundColor: 'rgba(255,65,105,1)',
            borderDash: [3, 3],
            borderWidth: 1,
            pointRadius: 0,
            pointHoverRadius: 2,
          },
          {
            label: 'Income',
            fill: 'start',
            data: [],
            pointBorderColor: 'rgba(0,123,255,1)',
            backgroundColor: 'rgba(0,123,255,0.1)',
            borderColor: 'rgba(0,123,255,1)',
            pointBackgroundColor: '#ffffff',
            pointHoverBackgroundColor: 'rgb(0,123,255)',
            borderWidth: 1.5,
            pointRadius: 0,
            pointHoverRadius: 3,
          },
        ],
      };

    beforeEach(() => {
        store = mockStore(initialState)
        //chartOptions={chartOptions}
    });

    it(' should render MonthlyBills correctly', () => {
        const wrapper = mount(
            <MonthlyBills 
                data={dummy_data}
                
            />
        );

        expect(wrapper).toMatchSnapshot();
    });
});