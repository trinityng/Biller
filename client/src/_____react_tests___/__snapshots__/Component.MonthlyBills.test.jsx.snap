// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MonthlyBills components  should render MonthlyBills correctly 1`] = `ReactWrapper {}`;
